components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -571,6 +614,190 @@ export function createWalletApi () {
             id: `${id}-${timeframe ?? BraveWallet.AssetPriceTimeframe.Live}`
           }))
       }),
+      getPriceHistory: query<TokenPriceHistory[], GetPriceHistoryArg>({
+        queryFn: async (arg, _api, _extraOptions, baseQuery) => {
+          try {
+            const {
+              data: { assetRatioService }
+            } = baseQuery(undefined)
+            const { success, values } = await assetRatioService.getPriceHistory(
+              arg.tokenParam,
+              arg.vsAsset,
+              arg.timeFrame
+            )
+
+            if (success && values) {
+              return {
+                data: values.map((value) => ({
+                  date: mojoTimeDeltaToJSDate(value.date),
+                  close: Number(value.price)
+                }))
+              }
+            }
+
+            return {
+              error: `Unable to fetch price history for token: ${arg.tokenParam}`
+            }
+          } catch (error) {
+            const message =
+              'Error getting price history: ' + error?.message ||
+              JSON.stringify(error)
+            console.error('Error getting price history: ', error)
+            return {
+              error: message
+            }
+          }
+        },
+        providesTags: (_result, err, arg) =>
+          err
+            ? ['PriceHistory']
+            : [
+                {
+                  type: 'PriceHistory',
+                  id: `${arg.tokenParam}-${arg.vsAsset}-${arg.timeFrame}`
+                }
+              ]
+      }),
+      getPricesHistory: query<TokenPriceHistory[], GetPricesHistoryArg>({
+        queryFn: async (
+          { tokens, vsAsset, timeframe, tokenBalancesRegistry },
+          { dispatch },
+          extraOptions,
+          baseQuery
+        ) => {
+          try {
+            const {
+              data: { assetRatioService }
+            } = baseQuery(undefined)
+
+            // dedupe tokens to prevent duplicate price history requests
+            const uniqueIds = [...new Set(tokens.map(getPriceIdForToken))]
+
+            const history = await mapLimit(
+              uniqueIds,
+              maxConcurrentPriceRequests,
+              async function (id: string) {
+                const { success, values } =
+                  await assetRatioService.getPriceHistory(
```

Comment:
should we query `getPriceHistory` here instead so the result is cached?

Reply:
@josheleonard We'll need a try...catch around the `getPriceHistory` query by the way. Should we also pass in a `forceRefetch: true` there?

Reply:
Switched back to the api proxy since I moved the query to a dedicated endpoint.